src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -507,6 +507,20 @@ export class TerminalService implements ITerminalService {
 		}
 	}
 
+	private async _refreshAvailableProfilesNow(): Promise<void> {
```

Comment:
We should use this in the throttled method

Reply:
Also about throttled method - probably it should be changed from `private async _refreshAvailableProfiles(): Promise<void>` to `private _refreshAvailableProfiles(): void` which would match decorated method signature/behavior.
TS at this moment doesn't override method signature with decorated one.

Reply:
@IllusionMH maybe you're right as it would reinforce that you should never await `_refreshAvailableProfiles`, we'd just have to avoid using async/await then in favor of `then`

Reply:
@Tyriar I'm not sure "we'd just have to avoid using async/await then in favor of then"  do you mean don't use `await` and use `.then` instead in this file?

Looks like it not using `await` on non-promise values can be enforced with https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/await-thenable.md so no need for `.then`

Reply:
right we weren't awaiting it before anyway, so no changes are required to change the signature

Reply:
Nvm, I was thinking if that change was made in isolation, diff looks good